File: MemoryGame/Models/AchievementModel.swift
//
//  AchievementModel.swift
//  Memory Match Kids
//

import Foundation

struct AchievementModel: Identifiable, Equatable {
    let id: String
    let title: String
    let description: String
    let icon: String
    let requiredStars: Int
    let requiredLevels: Int
}

extension AchievementModel {
    static let catalog: [AchievementModel] = [
        AchievementModel(id: "first_match", title: "First Match",
                         description: "Complete your first level",
                         icon: "star.fill", requiredStars: 0, requiredLevels: 1),
        AchievementModel(id: "five_levels", title: "Warming Up",
                         description: "Complete 5 levels",
                         icon: "flame.fill", requiredStars: 0, requiredLevels: 5),
        AchievementModel(id: "ten_levels", title: "On a Roll",
                         description: "Complete 10 levels",
                         icon: "bolt.fill", requiredStars: 0, requiredLevels: 10),
        AchievementModel(id: "perfectionist", title: "Perfect!",
                         description: "Earn 3 stars on a level",
                         icon: "checkmark.seal.fill", requiredStars: 0, requiredLevels: 1),
        AchievementModel(id: "rising_star", title: "Rising Star",
                         description: "Earn 25 stars",
                         icon: "star.circle.fill", requiredStars: 25, requiredLevels: 0),
        AchievementModel(id: "twentyfive_levels", title: "Memory Buff",
                         description: "Complete 25 levels",
                         icon: "books.vertical.fill", requiredStars: 0, requiredLevels: 25),
        AchievementModel(id: "star_collector", title: "Star Collector",
                         description: "Earn 100 stars",
                         icon: "sparkles", requiredStars: 100, requiredLevels: 0),
        AchievementModel(id: "perfect_gold", title: "Gold Champion",
                         description: "Earn 3 stars on 15 levels",
                         icon: "crown.fill", requiredStars: 0, requiredLevels: 15),
        AchievementModel(id: "speed_demon", title: "Speed Demon",
                         description: "Finish a timed level under 60 seconds",
                         icon: "stopwatch.fill", requiredStars: 0, requiredLevels: 0),
        AchievementModel(id: "memory_master", title: "Memory Master",
                         description: "Complete every level",
                         icon: "graduationcap.fill", requiredStars: 0,
                         requiredLevels: LevelCatalog.levelCount)
    ]
}

struct DailyChallengeState: Equatable {
    var dateKey: String
    var levelId: String
    var completed: Bool
}

File: android/app/src/main/java/com/memogame/app/model/AchievementModel.kt
package com.memogame.app.model

import com.memogame.app.data.LevelCatalog

data class AchievementModel(
    val id: String,
    val title: String,
    val description: String,
    val icon: String,
    val requiredStars: Int,
    val requiredLevels: Int
) {
    companion object {
        val catalog: List<AchievementModel> = listOf(
            AchievementModel(
                "first_match", "First Match", "Complete your first level",
                "star", 0, 1
            ),
            AchievementModel(
                "five_levels", "Warming Up", "Complete 5 levels",
                "flame", 0, 5
            ),
            AchievementModel(
                "ten_levels", "On a Roll", "Complete 10 levels",
                "bolt", 0, 10
            ),
            AchievementModel(
                "perfectionist", "Perfect!", "Earn 3 stars on a level",
                "seal", 0, 1
            ),
            AchievementModel(
                "rising_star", "Rising Star", "Earn 25 stars",
                "star_circle", 25, 0
            ),
            AchievementModel(
                "twentyfive_levels", "Memory Buff", "Complete 25 levels",
                "books", 0, 25
            ),
            AchievementModel(
                "star_collector", "Star Collector", "Earn 100 stars",
                "sparkles", 100, 0
            ),
            AchievementModel(
                "perfect_gold", "Gold Champion", "Earn 3 stars on 15 levels",
                "crown", 0, 15
            ),
            AchievementModel(
                "speed_demon", "Speed Demon", "Finish a timed level under 60 seconds",
                "stopwatch", 0, 0
            ),
            AchievementModel(
                "memory_master", "Memory Master", "Complete every level",
                "graduation", 0, LevelCatalog.LEVEL_COUNT
            )
        )
    }
}
